### PR TITLE
Statistical Functions(StdevP, StdevPT, VarP, VarPT) implementation.

### PR DESCRIPTION
**Pull request by omprakaash**
Implemented some statistical functions in the interpreter and included tests.
Edit: Used Welford's algorithm to implement VarianceAggreator. Is that okay or is there any other specific alg/implementation that is generally used for variance : https://en.wikipedia.org/wiki/Algorithms_for_calculating_variance#Welford's_online_algorithm.

Seems to avoid numerical issues and avoids doing two passes through the data to find the mean of the set first.

### CODE REVIEW

File: src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/TableMathfuncs.txt [hunk 1]
```diff
@@ -34,4 +34,49 @@ Blank()
 2
 
 >> Average([1, 2, 1/0, 4, 5], Value)
-#Error
\ No newline at end of file
+#Error
+
+>> VarP(1, 3, 5)
```
Comment (lesaltzm) [inline src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/TableMathfuncs.txt:39]: We're moving towards putting function-specific tests in their own .txt file. Makes searching easier, and it becomes kind of a spec for the function.

File: src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryMath.cs [hunk 2]
```diff
@@ -57,6 +57,58 @@ public virtual FormulaValue GetResult(IRContext irContext)
...
+
+            public virtual FormulaValue GetResult(IRContext irContext)
+            {
+                if (_count == 0)
+                {
+                    return CommonErrors.DivByZeroError(irContext);
+                }
+                else
+                {
+                    return new NumberValue(irContext, _m2Acc / _count);
```
Comment (lesaltzm) [inline src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryMath.cs:92]: Looks good!
I think NaN and +/- Inf as the result of the division should also return an error.

Comment (omprakaash) [reply src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryMath.cs]: Got it. Added the necessary checks.

File: src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/TableMathfuncs.txt (hunk 1 above)
Comment (omprakaash) [reply src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/TableMathfuncs.txt]: Moved the VarP and StdevP tests into their own files, but left sum and average alone as I noticed they have unused tests in the NotYetReady test-case folder.

Comment (lesaltzm) [reply src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/TableMathfuncs.txt]: Thanks. The existing ones will get cleaned up later, no need to worry about that for now. Just trying to make sure new ones are sensible.

File: src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryMath.cs (hunk 2 above)
Comment (lesaltzm) [reply src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryMath.cs]: We have a function for this, it's a pretty common case. Consider writing the return statement as 
`return FiniteChecker(irContext, 0, new NumberValue(irContext, _m2Acc/_count))`

Comment (omprakaash) [reply src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryMath.cs]: Modified the function to use FiniteChecker !

**Review by lesaltzm — APPROVED** [review]

**Review by MikeStall — APPROVED** [review]
:shipit:

**Comment by MikeStall** [comment]
> >> StdevP(1,2,3)

fwiw - I double checked that all these new tests do pass on the JS implementation. 

---
Refers to: src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/StdevP.txt:1 in f194fff. [](commit_id = f194fffcfde8f2b1ca3648a60c2766849da8b781, deletion_comment = False)

**Pull request merged**